src/rgw/rgw_datalog.cc

```diff
@@ -203,29 +204,18 @@ class RGWDataChangesOmap final : public RGWDataChangesBE {
 
 class RGWDataChangesFIFO final : public RGWDataChangesBE {
   using centries = std::vector<ceph::buffer::list>;
-  std::vector<std::unique_ptr<rgw::cls::fifo::FIFO>> fifos;
+  dynarray<LazyFIFO> fifos;
```

Comment:
nice improvement

Reply:
Thank you. dynarray is something I've been wishing for for a long time to solve the 'dynamically sized array of things containing mutexes' problem.